misk/src/main/kotlin/misk/logging/RateLimiter.kt

```diff
@@ -0,0 +1,120 @@
+package misk.logging
```

Comment:
This is copied from clientsync? Where are the tests?

Reply:
Yes I carried it over. Not using Guava RateLimiter because it was easier to test with this, simpler to interpret, and accomplishes what we need. Hope that is alright. 

Just added the tests as well.

Reply:
this could probably go in its own package or module. This was we can change clientsync to use the misk one and its available for other services as well